java/src/main/native/src/RmmJni.cpp

```diff
@@ -577,6 +578,80 @@ class pinned_fallback_host_memory_resource {
                            cuda::mr::host_accessible) noexcept
   {
   }
+
+#if CCCL_MAJOR_VERSION > 3 || (CCCL_MAJOR_VERSION == 3 && CCCL_MINOR_VERSION >= 1)
+
+ public:
+  /**
+   * @brief Allocates pinned host memory of size at least \p bytes bytes.
+   *
+   * @throws rmm::out_of_memory if the requested allocation could not be fulfilled due to to a
+   * CUDA out of memory error.
+   * @throws rmm::bad_alloc if the requested allocation could not be fulfilled due to any other
+   * reason.
+   *
+   * @param bytes The size, in bytes, of the allocation.
+   * @param alignment Alignment in bytes. Default alignment is used if unspecified.
+   *
+   * @return Pointer to the newly allocated memory.
+   */
+  static void* allocate_sync(std::size_t bytes, std::size_t alignment)
```

Comment:
All the newly added functions here cannot be `static`.
```suggestion
 void* allocate_sync(std::size_t bytes, std::size_t alignment)
```